Bottom: 02ba21569834d698cca7eb5533e92989da5b373e
Top:    550aee0dbc7845c62033d7b132057f7388ab2af4
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-06 16:44:29 +0000

Refresh of play-with-quad-edge-data

---

diff --git a/src/hid/common/edge3d.c b/src/hid/common/edge3d.c
index 3f08003..463c002 100644
--- a/src/hid/common/edge3d.c
+++ b/src/hid/common/edge3d.c
@@ -1,5 +1,5 @@
+#include <glib.h>
 #include <stdbool.h>
-#include <stdlib.h>
 
 #include "edge3d.h"
 
@@ -8,7 +8,7 @@ make_edge_info (bool is_stitch, bool is_round, double cx, double cy, double radi
 {
   edge_info *info;
 
-  info = malloc (sizeof(edge_info));
+  info = g_new0 (edge_info, 1);
   info->is_stitch = is_stitch;
   info->is_round = is_round;
   info->cx = cx;
@@ -21,5 +21,5 @@ make_edge_info (bool is_stitch, bool is_round, double cx, double cy, double radi
 void
 destroy_edge_info (edge_info *info)
 {
-  free (info);
+  g_free (info);
 }
diff --git a/src/hid/common/vertex3d.c b/src/hid/common/vertex3d.c
index 1704b2b..5445e4c 100644
--- a/src/hid/common/vertex3d.c
+++ b/src/hid/common/vertex3d.c
@@ -1,4 +1,4 @@
-#include <stdlib.h>
+#include <glib.h>
 
 #include "vertex3d.h"
 
@@ -10,7 +10,7 @@ make_vertex3d (double x, double y, double z)
 {
   vertex3d *v;
 
-  v = malloc (sizeof(vertex3d));
+  v = g_new0 (vertex3d, 1);
   v->x = x;
   v->y = y;
   v->z = z;
@@ -22,5 +22,5 @@ make_vertex3d (double x, double y, double z)
 void
 destroy_vertex3d (vertex3d *v)
 {
-  free (v);
+  g_free (v);
 }
